packages/@react-aria/datepicker/src/useDateField.ts

```diff
@@ -181,6 +181,9 @@ export function useDateField<T extends DateValue>(props: AriaDateFieldOptions<T>
         if (props.onKeyUp) {
           props.onKeyUp(e);
         }
+      },
+      style: {
+        unicodeBidi: 'isolate'
```

Comment:
added for DateRangePicker because things were looking a little funky without it. wraps around each datefield. see [codepen](https://codepen.io/yihuiliao/pen/OPLQWvv) for reproduction